packages/magento-wishlist/components/WishlistItemActionCard/WishlistItemActionCard.tsx

```diff
@@ -0,0 +1,269 @@
+  sx?: SxProps<Theme>
+  selectedOptions?: InputMaybe<string[]> | undefined
+  children?: React.ReactNode
+} & OwnerState &
+  OptionalProductWishlistParent &
+  Omit<ActionCardProps, 'value' | 'image' | 'price' | 'title' | 'action'>
+
+type OwnerState = { withOptions?: boolean }
+
+const compName = 'WishlistItemActionCard' as const
+const parts = [
+  'item',
+  'picture',
+  'badge',
+  'productLink',
+  'image',
+  'itemName',
+  'itemPrice',
+  'discountPrice',
+  'root',
+] as const
+const { classes } = extendableComponent<OwnerState, typeof compName, typeof parts>(compName, parts)
+export const productImageSizes = {
+  small: responsiveVal(60, 80),
+  medium: responsiveVal(60, 80),
+  large: responsiveVal(100, 120),
+}
+
+const typographySizes = {
+  small: 'body2',
+  medium: 'body1',
+  large: 'subtitle1',
+}
+
+export function WishlistItemActionCard(props: WishlistItemActionCardProps) {
+  const {
+    sku,
+    name,
+    url_key,
+    price_range,
+    small_image,
+    __typename: productType,
+    thumbnail,
+    children,
+    sx = [],
+    wishlistItemId,
+    size = 'large',
+    selectedOptions,
+    ...rest
+  } = props
+
+  const productLink = useProductLink({ url_key, __typename: productType })
+  const { cache } = useApolloClient()
+
+  const { loggedIn } = useCustomerSession()
+
+  const { data: GetCustomerWishlistData } = useCustomerQuery(GetIsInWishlistsDocument)
+
+  const [removeWishlistItem] = useMutation(RemoveProductFromWishlistDocument)
+
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null)
+  const open = Boolean(anchorEl)
+
+  const handleClick = <T extends HTMLElement>(event: React.MouseEvent<T>) => {
+    setAnchorEl(event.currentTarget)
+  }
+
+  const handleClose = <T extends HTMLElement>(event: React.MouseEvent<T>) => {
+    const t = event.target as T
+    if (t.id === 'remove') {
+      if (loggedIn) {
+        let itemIdToDelete = wishlistItemId
+
+        /** When no internal ID is provided, fetch it by sku */
+        if (!itemIdToDelete) {
+          const wishlistItemsInSession =
+            GetCustomerWishlistData?.customer?.wishlists[0]?.items_v2?.items || []
+
+          const item = wishlistItemsInSession.find(
+            (element) => element?.product?.url_key === url_key,
+          )
+          if (item?.id) {
+            itemIdToDelete = item.id
+          }
+        }
+
+        if (itemIdToDelete) {
+          // eslint-disable-next-line @typescript-eslint/no-floating-promises
+          removeWishlistItem({ variables: { wishlistItemId: itemIdToDelete } })
+        }
+      } else {
+        cache.modify({
+          id: cache.identify({ __typename: 'GuestWishlist' }),
+          fields: {
+            items(existingItems = []) {
+              const items = existingItems.filter((guestItem) => guestItem.sku !== sku)
+              return items
+            },
+          },
+        })
+      }
+    }
+    setAnchorEl(null)
+  }
```

Comment:
This code looks really sloppy and could easily be optimised. If you need help just let me know.